homeassistant/components/lyric/sensor.py

```diff
@@ -0,0 +1,251 @@
+        """Initialize Honeywell Lyric sensor."""
+
+        super().__init__(
+            coordinator,
+            location,
+            device,
+            f"{device.macID}_indoor_temperature",
+            "Indoor Temperature",
+            None,
+            DEVICE_CLASS_TEMPERATURE,
+            unit_of_measurement,
+        )
+
+    @property
+    def state(self) -> str:
+        """Return the state of the sensor."""
+        return self.device.indoorTemperature
+
+
+class LyricOutdoorTemperatureSensor(LyricSensor):
+    """Defines a Honeywell Lyric sensor."""
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        location: LyricLocation,
+        device: LyricDevice,
+        unit_of_measurement: str = None,
+    ) -> None:
+        """Initialize Honeywell Lyric sensor."""
+
+        super().__init__(
+            coordinator,
+            location,
+            device,
+            f"{device.macID}_outdoor_temperature",
+            "Outdoor Temperature",
+            None,
+            DEVICE_CLASS_TEMPERATURE,
+            unit_of_measurement,
+        )
+
+    @property
+    def state(self) -> str:
+        """Return the state of the sensor."""
+        return self.device.outdoorTemperature
+
+
+class LyricOutdoorHumiditySensor(LyricSensor):
+    """Defines a Honeywell Lyric sensor."""
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        location: LyricLocation,
+        device: LyricDevice,
+        unit_of_measurement: str = None,
+    ) -> None:
+        """Initialize Honeywell Lyric sensor."""
+
+        super().__init__(
+            coordinator,
+            location,
+            device,
+            f"{device.macID}_outdoor_humidity",
+            "Outdoor Humidity",
+            None,
+            DEVICE_CLASS_HUMIDITY,
+            "%",
+        )
+
+    @property
+    def state(self) -> str:
+        """Return the state of the sensor."""
+        return self.device.displayedOutdoorHumidity
+
+
+class LyricNextPeriodSensor(LyricSensor):
+    """Defines a Honeywell Lyric sensor."""
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        location: LyricLocation,
+        device: LyricDevice,
+        unit_of_measurement: str = None,
+    ) -> None:
+        """Initialize Honeywell Lyric sensor."""
+
+        super().__init__(
+            coordinator,
+            location,
+            device,
+            f"{device.macID}_next_period_time",
+            "Next Period Time",
+            None,
+            DEVICE_CLASS_TIMESTAMP,
+        )
+
+    @property
+    def state(self) -> datetime:
+        """Return the state of the sensor."""
+        device = self.device
+        time = dt_util.parse_time(device.changeableValues.nextPeriodTime)
+        now = dt_util.utcnow()
+        if time <= now.time():
+            now = now + timedelta(days=1)
+        return dt_util.as_utc(datetime.combine(now.date(), time))
```

Comment:
The timestamp should be isoformatted.

https://docs.python.org/3/library/datetime.html#datetime.datetime.isoformat